### PR TITLE
X11 backend

### PR DESCRIPTION
**Pull request by kekekeks**
This will only require libX11, so we won't depend on locally-installed GTK  version. 
It will also remove native-to-managed callbacks which make us swallow exceptions in GTK backend right now.

Limitation: File dialogs currently still depend on GTK running on a separate thread. Will be fixed by #2016

### CODE REVIEW

File: samples/PlatformSanityChecks/Program.cs [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+        });
+
+        static void CheckPlatformThreading()
+        {
+            CheckSignaling();
+            CheckTimerOrdering();
+            CheckTimerTicking();
+
+        }
+
```
Comment (RoadTrain) [inline samples/PlatformSanityChecks/Program.cs]: Probably unintended empty lines

File: src/Avalonia.X11/X11Atoms.cs [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+		public readonly IntPtr CLIPBOARD_MANAGER;
+		public readonly IntPtr SAVE_TARGETS;
+		public readonly IntPtr MULTIPLE;
+		public readonly IntPtr PRIMARY;
+		public readonly IntPtr OEMTEXT;
+		public readonly IntPtr UNICODETEXT;
+		public readonly IntPtr TARGETS;
+	    public readonly IntPtr UTF8_STRING;
+	    public readonly IntPtr UTF16_STRING;
+	    public readonly IntPtr ATOM_PAIR;
```
Comment (RoadTrain) [inline src/Avalonia.X11/X11Atoms.cs]: Mixed up tabs and spaces.

**Comment by kekekeks** [comment]
@danwalmsley Don't bother merging master here, the code still uses the old dialog API. I'll fix it later this month.

**Review by danwalmsley — APPROVED** [review]
LGTM - given this alot of testing on various applications and a few different distros.

**Pull request merged**